File: Homeworks/2. Web Design and UI technlogies/3. JavaScript-Part-1/09.Strings/02.Correct brackets/Correct brackets.js
/**  Correct brackets
 * Write a JavaScript function to check if in a given expression the brackets are put correctly.
 */
function correctBrackets(str) {
    var isCorrect = false,
        op = 0,
        cl = 0,
        str=str[0].trim();

    for (var i = 0; i < str.length; i += 1) {
        if (str[i] == '(') {
            op += 1;
        } else if (str[i] == ')') {
            cl += 1;
        }
    }

    if (op == cl) {
        isCorrect = true;
    }

    console.log(isCorrect ? 'Correct' : 'Incorrect');
}

correctBrackets([')(a+b))']);

File: Homeworks/1. Programming/2. C#-Part-2/06.Strings-and-Text-Processing/03.Correct brackets/CorrectBrackets.cs
//Write a program to check if in a given expression the(and ) brackets are put correctly.
namespace _03.Correct_brackets
{
    using System;

    class CorrectBrackets
    {
        static void Main()
        {
            var input = Console.ReadLine();
            int countOpen = 0, countClose = 0;


            //check how many times we have brackets
            for (int i = 0; i < input.Length; i++)
            {
                if (input[i] == '(')
                {
                    countOpen++;
                }
                else if (input[i] == ')')
                {
                    countClose++;
                }
            }
            //check if brackets are correct

            if (countOpen == countClose)
            {
                Console.WriteLine("Correct");
            }
            else
            {
                Console.WriteLine("Incorrect");
            }

        }
    }
}
